### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by PNdunguMaina**
In this project, I did the following: ⚒️ 

- [ ] Created model classes for all entities following the ERD provided.
- [ ] Set up associations between models.
- [ ] Used models to insert data via the Rails Console.
- [ ] Implemented CRUD methods
- [ ] Created custom methods in the models

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
+  private
+
+  def update_comments_count
+    update(CommentsCounter: comments.count)
+  end
```
Comment (danmainah) [inline app/models/post.rb:19]: - [ ] @PNdunguMaina great work in your methods, however this method is suppose to be in the comments model, kindly move it to the comment model.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def update_likes_count
+    update(LikesCounter: likes.count)
+  end
```
Comment (danmainah) [inline app/models/post.rb:23]: - [ ] @PNdunguMaina great work in your models, however, this method is supposed to be in the ike model, kindly move it to the like model.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @PNdunguMaina , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by AnselemOdims — APPROVED** [review]
### Hi  @PNdunguMaina 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**